autogpt_platform/backend/backend/util/service.py

```diff
@@ -170,31 +185,53 @@ def __init__(self):
             logger.debug(f"Successfully connected to service [{service_name}]")
 
         def __getattr__(self, name: str) -> Callable[..., Any]:
-            return getattr(self.proxy, name)
+            res = getattr(self.proxy, name)
+            return res
 
     return cast(AS, DynamicClient())
 
 
 # --------- UTILITIES --------- #
 
+builtin_types = [*vars(builtins).values(), NoneType, Enum]
+
 
 def _pydantic_models_from_type_annotation(annotation) -> Iterator[type[BaseModel]]:
     # Peel Annotated parameters
     if (origin := get_origin(annotation)) and origin is Annotated:
         annotation = get_args(annotation)[0]
 
-    if origin := get_origin(annotation):
-        if origin is UnionType:
-            types = get_args(annotation)
-        else:
-            types = [origin]
+    origin = get_origin(annotation)
```

Comment:
I rewrote this pydantic peeling logic since we are not doing it recursively. cc @Pwuts